source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACOHostConfig.mm

```diff
@@ -365,28 +365,54 @@ + (UIColor *)convertHexColorCodeToUIColor:(const std::string&)hexColorCode
     return color;
 }
 
-- (UIColor *)getBackgroundColorForContainerStyle:(ACRContainerStyle)style
+- (ContainerStyleDefinition) getColorPalette:(ACRContainerStyle)style
```

Comment:
you are returning a struct by value. i suggest adding another param to diff the request either to foreground or background and return only those string instead of whole structure.

Reply:
The problem is that the background is a `string` whereas the foreground is a `ColorsConfig` struct. Having this helper method lets me get the background through `getBackgroundColorForContainerStyle()` (in an appropriately converted format) and the foreground colors directly.

Reply:
can you make it return reference?

---
In reply to: [249862786](https://github.com/Microsoft/AdaptiveCards/pull/2289#discussion_r249862786) [](ancestors = 249862786)

Reply:
can get rid of this fuction by making getTextBlock color an instance method and access color config through its host config and add optional style param.

---
In reply to: [249968877](https://github.com/Microsoft/AdaptiveCards/pull/2289#discussion_r249968877) [](ancestors = 249968877,249862786)